SimPEG/EM/TDEM/ProblemTDEM.py

```diff
@@ -719,7 +878,62 @@ def getRHSDeriv(self, tInd, src, v, adjoint=False):
         # right now, we are assuming that s_e, s_m do not depend on the model.
         return Utils.Zero()
 
+    def getInitialFields(self):
+        """
+        Ask the sources for initial fields
+        """
 
+        Srcs = self.survey.srcList
+
+        if self._fieldType in ['b', 'j']:
+            ifields = np.zeros((self.mesh.nF, len(Srcs)))
+        elif self._fieldType in ['e', 'h']:
+            ifields = np.zeros((self.mesh.nE, len(Srcs)))
+        if self.verbose:
+            print ("Calculating Initial fields")
+
+        for i, src in enumerate(Srcs):
+            # Check if the source is grounded
+            if src.srcType == "Galvanic" and src.waveform.hasInitialFields:
+                # Check self.Adcinv and clean
+                if self.Adcinv is not None:
+                    self.Adcinv.clean()
+                # Factorize Adc matrix
+                if self.verbose:
+                    print ("Factorize system matrix for DC problem")
+                Adc = self.getAdc()
+                self.Adcinv = self.Solver(Adc)
+
+            ifields[:, i] = (
+                ifields[:, i] + getattr(
+                    src, '{}Initial'.format(self._fieldType), None
+                )(self)
+            )
+
+        return ifields
+
+    def getAdc(self):
```

Comment:
just a thought (can be addressed in a separate pr), can we use the infrastructure already in the DC problem to achieve this instead of repeating?